kep/42-podgroup-coscheduling/README.md

```diff
@@ -135,12 +141,75 @@ cluster upgrade.
 
 #### QueueSort
 
-To make sure a group of pods can be scheduled as soon as possible. We implemented this `extension point`. The main progress is as follows:
+For Pods belong to a same podGroup should be popped out of the scheduling queue one by one,
+which requires an efficient and wise algorithm. This is quite different with plain Pods,
+so we need to reimplement the QueueSort plugin.
 
-1. Sort based on Pod Priority
-2. Pod does not belong to any group
-3. PodGroup creation time
-4. Pod creation time
+Basically, we need to watch for all the podGroups and maintain a cache for them
+in coscheduling's core package:
+
+- when podGroup created, we'll create a `queuedPodGroup` in the cache
+- when podGroup updated, we'll do nothing right now
+- when podGroup deleted, we'll clean up the objet in the cache
+
+The structure looks like below:
+
+```golang
+type PodGroupManager struct {
+ // ...
+ // <new>
+ // key is the podGroup name, value is the queued podGroup info.
+ podGroups map[string]*queuedPodGroup
+}
+
+// queuedPodGroup's lifecycle is synchronized with the podGroup object.
+type queuedPodGroup struct {
+ // Timestamp is the podGroup's queued time.
+ // - timestamp will be initialized when the first pod enqueues
+ // - timestamp will be renewed when a pod re-enqueues for failing the scheduling
```

Comment:
could you reword a bit?

do you mean "... renewed once a pod that belongs to a previously failed podGroup gets re-enqueued"?